Fix:  parse NDJSON correctly, fixes #825

MariaGolubev opened the pull request:
Here are the fixes I propose. I'm not entirely sure everything is correct, so please review the code.
[ #825](https://github.com/0xPlaygrounds/rig/issues/825)

joshua-mo-143 commented:
Hi Maria!

We will probably need to revert to the previous implementation that was implemented prior to the update. You can find the original implementation here: https://github.com/0xPlaygrounds/rig/blob/9a3a5ef4532c2b75822d8001c1e20f44c40d3aab/rig-core/src/providers/ollama.rs

MariaGolubev commented:
Overall, Ollama is compatible with OpenAI. It might support SSE through /v1/chat/completions, but I haven’t checked that. However, the standard protocol for Ollama is NDJSON.

MariaGolubev commented:
Oh, I didn’t read your message. I added the correct error type handling. Before, `EmbeddingError::ProviderError()`  was used everywhere, which is sometimes incorrect. I also added tracing debug information so that errors in services like [Jaegert](https://www.jaegertracing.io/)  can be tracked.

```sh
2025-09-18T16:17:19.468126Z ERROR rig: Failed to send request to Ollama /api/embed: error sending request for url (http://localhost:11434/api/embed)
Error: HttpError: error sending request for url (http://localhost:11434/api/embed)

Caused by:
    0: error sending request for url (http://localhost:11434/api/embed)
    1: client error (Connect)
    2: tcp connect error
    3: Connection refused (os error 111)
```
```sh
2025-09-18T16:15:11.846743Z ERROR rig: Ollama returned error status 404 Not Found with body: {"error":"model \"llama3.2 \" not found, try pulling it first"}
Error: ProviderError: {"error":"model \"llama3.2 \" not found, try pulling it first"}
```
